### ISSUE TITLE
If I want to make attention in background.

### ISSUE THREAD

**Issue by deep0learning** [yhlleo/uaggan#2]
Hi,
Thank you for this task. If I want to make attention in the background that I need to change rather than the object. For example, in domain A and B, we have horse images,  when translating from domain A - B, I want to keep the same horse in domain B but the background will be changed as domain A. How I can do that? Thank you in advance.

**Comment by yhlleo** [comment]
Perhaps, you can revise the attention mask by (1-mask) to get the mask of background. @deep0learning

**Comment by deep0learning** [comment]
Hi,

Thank you so much. I have changed the following two lines, but I did not get the desired result.

`        self.masked_fake_B = self.fake_B*(1 - self.att_A) + self.real_A*self.att_A #change line no 123`
`        self.masked_fake_A = self.fake_A*(1 - self.att_B) + self.real_B*self.att_B #change line no 129`

**Comment by yhlleo** [comment]
Perhaps, the style of background is not clear in this task. Specially, revising the code from 
```
self.masked_fake_B = self.fake_B*self.att_A + self.real_A*(1-self.att_A)
```
into 
```
self.masked_fake_B = self.fake_B*(1 - self.att_A) + self.real_A*self.att_A
```
makes no differences for the final results. The key problem is that what your domains are dominated by. In fact, the attention module learns the common visual patterns of each domain. So, if the domain is dominated by the similar background in your case, the attention module would learn this. @deep0learning

**Comment by deep0learning** [comment]
The style of background is clear in that case. Should I change in other codes as well? Did I miss anything?

**Comment by yhlleo** [comment]
Does the foreground is clear as well in your case? I don't know your project well. If you'd like to discuss about it in private, you can explain your project and share some examples by sending an email to me. @deep0learning

**Comment by deep0learning** [comment]
Thank you so much for your reply.
I am trying to sort out the problem. Actually, I have applied this code on the horse to zebra dataset where my intention was to translate the background. But, I did not get the desired result, though the paper suggested that "If s_a attends to an image region without a relevant foreground instance to translate, then the result s^ will preserve its source domain class (i.e. a horse will remain a horse)"
